src/common/darktable.c

```diff
@@ -1530,11 +1544,25 @@ int dt_init(int argc, char *argv[], const gboolean init_gui, const gboolean load
   }
   free(config_info);
 
-  // last but not least construct the popup that asks the user about
-  // images whose xmp files are newer than the db entry
-  if(init_gui && changed_xmp_files)
+  if(init_gui)
   {
-    dt_control_crawler_show_image_list(changed_xmp_files);
+    // start the thumbnail backthread only if using the default db
+    const gboolean backthumb =
+        dt_conf_get_bool("cache_disk_backend")
```

Comment:
First try and didn't start. I had a new DB with 25 pictures. I have changed the setting to 4k and was waiting for the thumbs to be generated. Nothing happened and I found out why. The thread is started only if `cache_disk_backend` was set. This is a constraint that should be lifted to me. Two options, remove this check here or attach to the signal `DT_SIGNAL_PREFERENCES_CHANGE` to stop/start the job.

Reply:
Sure. But I thought "respect if user does not want to write thumbs". No?

Reply:
Maybe I wasn't clear, and yes I agree with you. My idea was:
1. Start the job even if `cache_disk_backend` if false
2. Exit the job early if `cache_disk_backend` if false
3. Check again if the job has to be started each time `DT_SIGNAL_PREFERENCES_CHANGE` is raised.

Now, I see even a better option, just implement 3 above. Thought?

Reply:
i put the check for writing into the backthread idle loop.

I could rewrite the whole job-parameters usage controlled by the `DT_SIGNAL_PREFERENCES_CHANGE` with some additional work as there are some preference settings involved but i personally think that's not necessary as the footprint is very low to me. Let me know if you disagree here.

Reply:
Looks good to me after testing. Indeed this has a very low impact on the perf.